### PR TITLE
Instrument buffer after contract state events stream

### PR DESCRIPTION
**Pull request by tudor-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala [hunk 1]
```diff
@@ -318,14 +318,16 @@ private[appendonlydao] final class TransactionsReader(
...
+    val transactionLogUpdatesSource = groupContiguous(eventsSource)(by = _.transactionId)
+      .map { v =>
+        val tx = toTransaction(v)
+        (tx.offset, tx.events.last.eventSequentialId) -> tx
+      }
+      .mapMaterializedValue(_ => NotUsed)
+
     InstrumentedSource
       .bufferedSource(
-        original = groupContiguous(eventsSource)(by = _.transactionId)
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:323]: Unrelated to this PR: aestethics refactoring

Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:323]: As long as you preserve `groupContiguous(eventsSource)(by = _.transactionId)`. I don't know who wrote but the original author clearly has a great taste. :trollface:

**Review by stefanobaghino-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala (hunk 1 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:323]: Original author :trollface:

Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:323]: >As long as you preserve groupContiguous(eventsSource)(by = _.transactionId). I don't know who wrote but the original author clearly has a great taste. 

@stefanobaghino-da Or actually, wdym? Maybe I'm missing something

Comment (stefanobaghino-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala:323]: Just kidding. 😉 LGTM.

**Review by mziolekda — APPROVED** [review]
Looks good

**Pull request merged**